### PR TITLE
don't overwrite user config

### PR DESCRIPTION
**Pull request by cacheonly**
Hello @adrianlee44, @sadappa, 

Please review the following commits I made in branch 'dont-overwrite-config'.

2e960670bfbb3b1221e69f89970635c700ad10a5 (2017-12-13 16:27:21 -0800)
don't overwrite user config

R=@adrianlee44
R=@sadappa

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ var generateToken = exports.generateToken = function generateToken(config, cb) {
 
     var http_options = {
         schema: config.schema || configuration.default_options.schema,
-        host: utils.getDefaultApiEndpoint(config.mode) || config.host || configuration.default_options.host,
+        host: config.host || utils.getDefaultApiEndpoint(config.mode) || configuration.default_options.host,
```
Comment (sadappa) [inline lib/api.js:62]: hmm is it a common pattern to fork service libraries in such cases?

Comment (adrianlee44) [reply lib/api.js:62]: have you thought of making a PR for this?

Comment (cacheonly) [reply lib/api.js:62]: @sadappa for the sake of time I'm forking the library. @adrianlee44 yes but I'm gonna file an issue and make sure it's a regression/bug

Comment (cacheonly) [reply lib/api.js:62]: https://github.com/paypal/PayPal-node-SDK/issues/326

**Pull request merged**